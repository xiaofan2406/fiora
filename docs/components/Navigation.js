/* @flow */
import React from 'react';
import { css } from 'react-emotion';
import { NavLink } from 'react-router-dom';
import { theme, spacing } from 'styles';
import { NAV_LINKS } from 'utils/constants';

const cssNavigation = css`
  position: fixed;
  left: ${spacing.unit}px;
  bottom: ${spacing.unit}px;
  display: flex;
  flex-direction: column;
  border: 1px solid ${theme.borderColor};
  & > .link {
    text-decoration: none;
    padding: 0.5em;
    display: inline-block;
    &:hover {
      background-color: ${theme.borderColor};
    }
    &.active {
      border-right: 2px solid ${theme.primaryColor};
    }
  }
`;

const Navigation = () => (
  <div className={cssNavigation}>
    {NAV_LINKS.map(link => (
      <NavLink
        className="link"
        activeClassName="active"
        key={link.to}
        exact={link.exact}
        to={link.to}
      >
        {link.name}
      </NavLink>
    ))}
  </div>
);

export { Navigation as Component };

export default Navigation;
