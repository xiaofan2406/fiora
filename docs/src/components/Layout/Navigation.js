import React from 'react';
import { css } from 'emotion';
import { NavLink } from 'react-router-dom';
import { ROUTES } from 'configs';
import { theme, variables } from 'styles';

const navLink = css`
  color: ${theme.inverseColor};
  text-decoration: none;
  padding: 0 0.5em;
  display: inline-block;
  line-height: ${variables.Layout.headerHeight}px;
  height: ${variables.Layout.headerHeight}px;
  &:hover {
    background-color: ${theme.bgAccentColor};
  }
`;

const navLinkActive = css`
  border-bottom: 2px solid ${theme.primaryColor};
`;

function Navigation() {
  return (
    <div>
      {Object.values(ROUTES).map(route => (
        <NavLink
          className={navLink}
          activeClassName={navLinkActive}
          key={route.path}
          exact={route.exact}
          to={route.path}
        >
          {route.name}
        </NavLink>
      ))}
    </div>
  );
}

export { Navigation as Component };

export default Navigation;
