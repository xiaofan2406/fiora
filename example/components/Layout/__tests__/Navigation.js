import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { Component } from '../Navigation';

it('should match snapshot', () => {
  const props = { classes: {} };
  const component = shallow(<Component {...props} />);

  expect(toJson(component)).toMatchSnapshot();
});
