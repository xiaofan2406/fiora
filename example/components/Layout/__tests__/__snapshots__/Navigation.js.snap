// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <NavLink
    activeClassName="css-rg4nls"
    ariaCurrent="true"
    className="css-1xa38ke"
    exact={true}
    key="/"
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="css-rg4nls"
    ariaCurrent="true"
    className="css-1xa38ke"
    key="/about"
    to="/about"
  >
    About
  </NavLink>
  <NavLink
    activeClassName="css-rg4nls"
    ariaCurrent="true"
    className="css-1xa38ke"
    key="/contact"
    to="/contact"
  >
    Contact
  </NavLink>
</div>
`;
